language/move-vm/runtime/src/data_cache.rs

```diff
@@ -87,6 +90,10 @@ impl<'r, 'l, S: MoveResolver> TransactionDataCache<'r, 'l, S> {
 
             let mut resources = BTreeMap::new();
             for (ty, (layout, gv)) in account_data_cache.data_map {
+                if check_store {
```

Comment:
Is this checking for `store` transitively, or only for `gv`? I.e., if I have something like:

```
struct HotPotato {}
struct X has store { h: HotPotato }
struct G has key { x: X }

fun try_move_to(account: &signer, g: G) {
  move_to(account, G)
}
```

will, this check discover the hot potato?

Reply:
You are right. I think we need to check the ability safety when packing elements.